Tests/Unit/Subscribers/ServiceContainerTest.php

```diff
@@ -106,6 +107,10 @@ public function testOnAutoCategoryResolver()
                 [ProductToRemoteCategory::class, $this->createMock(ProductToRemoteCategoryRepository::class)],
             ]));
 
+        $this->diContainer
+            ->method('get')
```

Comment:
better add ```->with('CategoryDenormalization')```. right now will return CategoryDenormalization always, doesn't matter what's the argument.